spec/controllers/settings_controller_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+
+RSpec.describe SettingsController, type: :controller do
+  render_views
+  let(:user) { FactoryGirl.create(:user) }
+
+  describe '#update_profile' do
+    let(:username) { 'miyukki' }
+    let(:password) { 'p@ssw0rd' }
+    let(:student_id) { 71349640 }
+    let(:ldap_bind_result) { true }
+    let(:ldap_info_result) {
+      double(:hash, {
+        uid: [username],
+        gecos: ["Yusei Yamanaka, SFC, #{student_id}, t13964yy@sfc.keio.ac.jp"],
+        uidnumber: ['11032'],
+        gidnumber: ['1000'],
+      })
+    }
+    before do
+      session[:user_id] = user.id # logged in
+      allow_any_instance_of(Net::LDAP).to receive(:bind).and_return(ldap_bind_result)
+      allow_any_instance_of(Net::LDAP).to receive_message_chain(:search, :first).and_return(ldap_info_result)
+      patch :update_profile, ldap: { username: username, password: password }
+    end
+
+    context 'when correct credential' do
+      context 'and receive valid old ldap user result' do
+        let(:ldap_info_result) {
+          double(:hash, {
+            uid: [username],
+            gecos: ['miyukki,,,'],
+            uidnumber: ['11032'],
+            gidnumber: ['1000'],
+          })
```

Comment:
Indent the right brace the same as the first position after the preceding left parenthesis.